pkg/discovery/monitoring/kubelet/kubelet_client.go

```diff
@@ -0,0 +1,88 @@
+package kubelet
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	stats "k8s.io/kubernetes/pkg/kubelet/apis/stats/v1alpha1"
+	kClient "k8s.io/kubernetes/pkg/kubelet/client"
+
+	"github.com/turbonomic/kubeturbo/pkg/discovery/util/httputil"
+
+	cadvisorapi "github.com/google/cadvisor/info/v1"
+)
+
+const (
+	summaryPath string = "/stats/summary/"
+	specPath    string = "/spec"
+)
+
+type Host struct {
+	IP       string
+	Port     int
+	Resource string
+}
+
+type kubeletClient struct {
+	config *kClient.KubeletClientConfig
+	client *http.Client
+}
+
+func (kc *kubeletClient) GetPort() int {
+	return int(kc.config.Port)
+}
+
+func (kc *kubeletClient) GetSummary(host Host) (*stats.Summary, error) {
+	requestURL := url.URL{
+		Scheme: "http",
+		Host:   fmt.Sprintf("%s:%d", host.IP, host.Port),
+		Path:   summaryPath,
+	}
+	if kc.config != nil && kc.config.EnableHttps {
+		requestURL.Scheme = "https"
+	}
+
+	req, err := http.NewRequest("GET", requestURL.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	summary := &stats.Summary{}
+	client := kc.client
+	err = httputil.PostRequestAndGetValue(client, req, summary)
+	return summary, err
+}
+
+func (kc *kubeletClient) GetMachineInfo(host Host) (*cadvisorapi.MachineInfo, error) {
```

Comment:
It is not working for Openshift 3.4 cluster.

<img width="738" alt="screen shot 2017-07-05 at 1 01 45 pm" src="https://user-images.githubusercontent.com/27221807/27875715-31f9137c-6182-11e7-8123-180722574aa2.png">